docs/BackupAndRestore.md

```diff
@@ -0,0 +1,594 @@
+...
+```
+
+The archive data before restore will be saved for house-keeping in a `/var/opt/nuodb/archive/<db>-save-<timestamp>` directory.
+It is recommended to move this directory after a successful database restore.
+
+One of the SMs serving a restored archive will be selected as the database restore coordinator, which will be responsible for preparing the domain state for the database in-place restore operation.
+All TEs will wait for the database restore to complete before starting, hence we can check their log entries which will show a successful database restore similar to the log snippet below:
+
+```
+2021-02-19T13:06:38.019+0000 INFO  root Waiting for database restore to complete ...
+2021-02-19T13:06:56.019+0000 INFO  root Found coordinator process startId=11, hostname=sm-database-nuodb-cluster0-demo-hotcopy-1.demo.nuodb.svc.cluster.local, archiveId=6
+2021-02-19T13:06:56.019+0000 INFO  root Waiting for coordinator process startId=11 to become RUNNING
+2021-02-19T13:06:57.019+0000 INFO  root Database demo restored by process startId=11, address=sm-database-nuodb-cluster0-demo-hotcopy-1/172.17.0.12:48006, archiveId=6
+...
+```
+
+The database restore request will be cleared after a successful database in-place restore and can be viewed using `nuodocker get restore requests --db-name <db>`.
+The database state should be manually verified by `nuocmd show domain` and using SQL queries to ensure that it's in the desired state after a successful restore.
+
+If there is a need to modify the database restore request, a new database shutdown and restart cycle is needed so that the request can be completed.
+The same applies if the reason for the previously failed database restore is corrected.
+When troubleshooting distributed database restore failures, it's important to collect logs from all database processes and their corresponding pods.
+
+> **NOTE**: In multi-cluster deployments, different backup cron jobs execute with different schedules unless they are configured to be part of the same backup group.
+
+Backup sets produced by different backup groups can't be used together to perform _distributed database in-place restore_.
+To perform a database restore in multi-cluster deployment, proceed with one of the approaches described below.
+
+1. [Select the hot copy SMs in one of the clusters](#fine-grained-archive-selection) and provide a backup set available in that cluster or `:group-latest` as `restore.source`.
+2. Make the selected backup set available as a remote URL in both clusters. It can be used as a restore source for all the SMs in the database only if no custom storage groups are configured.
+3. Configure HC SMs in all clusters to be part of a single backup group and leave one set of backup jobs enabled in only one of the clusters to control this backup group.
+This will ensure that all HC SMs backups will be coordinated during hot copy requests.
+
```

Comment:
This seems kind of awkward and kind of defeating the purpose of the HC SM StatefulSet. The user has to manually choose a specific backup set to make available remotely, presumably from among the HC SMs in both clusters. Can we achieve the same thing by issuing a restore request that excludes the HC SMs in one of the two clusters, so that it waits for the HC SMs in the other cluster to restore the database before joining?

Reply:
Option 1 will be used by default --- e.g specific backup set or `:group-latest` will restore only HC SMs in one of the clusters. The rest of the SMs will sync.

I decided to include other options in case, the user wants to leverage an accelerated sync use case for all SMs in the database.